src/core/services/AnimationService.ts

```diff
@@ -0,0 +1,6 @@
+
+
+export const animationFramePromise = () => new Promise((res) => requestAnimationFrame(res))
```

Comment:
These are helpers to allow us to use async/await on the requestAnimationFrame  and setTimeout functions for better looking code rather than chaining nested callbacks